back/app/validators/multiloc_validator.rb

```diff
@@ -1,72 +1,54 @@
-      record.errors.add(attribute, :values_not_all_strings, message: "non-string values (for #{locales}) cannot be accepted. Either the key should be removed, or the value should be replaced by an empty string") 
+  def validate_values_not_all_strings(record, attribute, value)
+    if value&.values && !value.values.all? { |v| v.is_a? String }
+      locales = value.keys.reject { |l| value[l].is_a?(String) }
+      message = "non-string values (for #{locales}) cannot be accepted. Either the key should be removed, or the value should be replaced by an empty string"
+      record.errors.add attribute, :values_not_all_strings, message: message
     end
   end
 
-  def validate_supported_locales record, attribute, value
+  def validate_supported_locales(record, attribute, value)
     locales = CL2_SUPPORTED_LOCALES.map(&:to_s)
     if !(value.keys - locales).empty?
-      record.errors.add(attribute, :unsupported_locales, 
-        message: (options[:message] || "contains unsupported locales #{(value.keys - locales)}"))
+      message = options[:message] || "contains unsupported locales #{value.keys - locales}"
+      record.errors.add attribute, :unsupported_locales, message: message
     end
   end
 
-  def validate_length record, attribute, value
-    if options[:length]
-      options[:length].each do |constraint, constraint_value|
-        if constraint == :in
-          if !value.values.select{ |v| v.size < constraint_value.first }.empty?
-            record.errors.add(attribute, :too_short,
-              message: (options[:too_short] || "is too short (minimum is #{constraint_value} characters)"))
-          end
-          if !value.values.select{ |v| v.size > constraint_value.last }.empty?
-            record.errors.add(attribute, :too_long,
-              message: (options[:too_long] || "is too long (maximum is #{constraint_value} characters)"))
-          end
-        elsif constraint == :is
-          if !value.values.select{ |v| v.size != constraint_value }.empty?
-            record.errors.add(attribute, :wrong_length,
-              message: (options[:wrong_length] || "is the wrong length (should be #{constraint_value} characters)"))
-          end
-        elsif constraint == :maximum
-          if !value.values.select{ |v| v.size > constraint_value }.empty?
-            record.errors.add(attribute, :too_long,
-              message: (options[:too_long] || "is too long (maximum is #{constraint_value} characters)"))
-          end
-        elsif constraint == :minimum
-          if !value.values.select{ |v| v.size < constraint_value }.empty?
-            record.errors.add(attribute, :too_short,
-              message: (options[:too_short] || "is too short (minimum is #{constraint_value} characters)"))
-          end
+  def validate_html(record, attribute, value)
+    if options[:html]
+      value.each do |key, html|
+        doc = Nokogiri::HTML.fragment html
+        next if doc.text == html
```

Comment:
I think it would be better to give the condition a name, because the meaning of `doc.text == html` may be unclear for the reader who does not know what `text` actually does.

```suggestion
        plain_text = doc.text == html
        next if plain_text
```